### PR TITLE
api clean up

### PR DESCRIPTION
**Pull request by alexkn**
TODO:
- [x] Roborock Implementation 
- [x] Viomi Implementation

### CODE REVIEW

File: lib/devices/MiioVacuum.js [hunk 1]
```diff
@@ -70,49 +69,84 @@ class MiioVacuum {
...
+     * @returns {Promise<void>}
+     */
+    async pollStatus() {
+        throw new NotImplementedError();
+    }
 
     /**
      * Returns the current status of the robot
+     *
+     * @public
```
Comment (rumpeltux) [inline lib/devices/MiioVacuum.js:104]: isn’t public the default already?

File: lib/devices/MiioVacuum.js [hunk 2]
```diff
@@ -309,18 +365,36 @@ class MiioVacuum {
     }
 
     /**
+     * @typedef AppLocale
+     * @property {string} name
+     * @property {string} bom
```
Comment (rumpeltux) [inline lib/devices/MiioVacuum.js:370]: what’s that?

File: lib/devices/MiioVacuum.js [hunk 3]
```diff
@@ -309,18 +365,36 @@ class MiioVacuum {
...
+     * @returns {Promise<AppLocale>}
      */
     async getAppLocale() {
         throw new NotImplementedError();
     }
 
+    /**
+     * @typedef Timer
+     * @property {string} id
+     * @property {string} cron
```
Comment (rumpeltux) [inline lib/devices/MiioVacuum.js:388]: I assume that’s the timer-spec in crontab format?

File: lib/webserver/ApiRouter.js [hunk 4]
```diff
@@ -27,6 +27,7 @@ const removeOldUploads = async (extension = null) => {
  * Api routes used by the webinterface
  *
  * @param {import("./WebServer")} webserver
+ * @returns {import("express").Router}
```
Comment (rumpeltux) [inline lib/webserver/ApiRouter.js:30]: Type is already imported so `@returns {express.Router}` should just work, right?

File: lib/devices/MiioVacuum.js (hunk 1 above)
Comment (alexkn) [reply lib/devices/MiioVacuum.js:104]: Yes, public is the default. I simply added the access modifiers everywhere to make it clearer.

File: lib/devices/MiioVacuum.js (hunk 2 above)
Comment (alexkn) [reply lib/devices/MiioVacuum.js:370]: I don't know.
Roborock returns 'bom': 'A.03.0005'

File: lib/devices/MiioVacuum.js (hunk 3 above)
Comment (alexkn) [reply lib/devices/MiioVacuum.js:388]: Yes

File: lib/webserver/ApiRouter.js (hunk 4 above)
Comment (alexkn) [reply lib/webserver/ApiRouter.js]: that's right

**Pull request merged**